src/lib/components/Channel/Stream/StreamControls.svelte

```diff
@@ -399,22 +405,31 @@
 		</button>
 
 		{#if $is_feature_restream_enabled}
-			<button
-				class="btn text-white border-none tooltip font-normal normal-case {$is_sharing_obs
-					? 'btn-primary'
-					: 'btn-neutral'}"
-				data-tip="Restream"
-				on:click={() => {
-					$is_sharing_obs = !$is_sharing_obs
-				}}
-				disabled={$is_sharing_screen ||
-					$is_sharing_webcam ||
-					$is_sharing_audio ||
-					!isHostOrGuest ||
-					!isChannelSocketConnected ||
-					!videoItemIsActive}>
-				<IconRestream />
-			</button>
+			<div class="drawer drawer-end">
+				<input id="restream-drawer" type="checkbox" class="drawer-toggle" />
+				<div class="drawer-content">
+					<!-- Page content here -->
+					<!-- svelte-ignore a11y-click-events-have-key-events -->
+					<label
+						for="restream-drawer"
+						class="btn text-white border-none tooltip font-normal normal-case items-center flex {restream_drawer
+							? 'btn-primary'
+							: 'btn-neutral'}"
+						data-tip="Restream"
+						on:click={() => {
+							restream_drawer = !restream_drawer
+						}}>
+						<IconRestream />
+					</label>
+				</div>
+				<!-- svelte-ignore a11y-click-events-have-key-events -->
+				<div class="drawer-side z-50" on:click={overlayClick}>
+					<label for="restream-drawer" aria-label="close sidebar" class="drawer-overlay" />
+					<ul class="menu p-4 w-96 drawer-height bg-base-200 text-base-content m-4 rounded-lg">
+						<DrawerRestream />
```

Comment:
Move DrawerRestream.svelte container to channel page.svelte (where DrawerChat.svelte is).
StreamControls is only for the controls.